client/src/diff/TextDiff.js

```diff
@@ -76,13 +76,16 @@ const process_indicators = (matched_indicators, indicator_status) => {
       if (indicator_pair.length === 2){
         const name_diff = Diff.diffWords(indicator_pair[0].name, indicator_pair[1].name);
         const methodology_diff = window.is_a11y_mode ? Diff.diffSentences(indicator_pair[0].methodology, indicator_pair[1].methodology) : Diff.diffWords(indicator_pair[0].methodology, indicator_pair[1].methodology);
-        const target_diff = Diff.diffWords(indicator_target_text(indicator_pair[0], false), indicator_target_text(indicator_pair[1], false));
+        const target_diff = Diff.diffWords(indicator_target_text(indicator_pair[0], false), indicator_target_text(indicator_pair[1], false));        
         const target_explanation_diff = Diff.diffWords(indicator_pair[0].target_explanation || "", indicator_pair[1].target_explanation || "");
-        const status = _.max([name_diff.length, methodology_diff.length, target_diff.length]) > 1 ?
-          (target_diff.length > 1 ?
+
+        const status = _.max([target_diff.length, target_explanation_diff.length]) > 1 && _.max([name_diff.length, methodology_diff.length]) > 1 ?
+            "indicator_desc_and_target_changed" :
+            _.max([target_diff.length, target_explanation_diff.length]) > 1 ?
             "target_changed" :
-            "indicator_desc_changed") :
-          "no_diff";
+            _.max([name_diff.length, methodology_diff.length]) > 1 ?
+            "indicator_desc_changed" :
+            "no_diff";
```

Comment:
Any ideas how it should be filtered with that change? & when getting status flag. Problem with this is that it'd need to loop through the status object every time it needs to read from it.

Reply:
I mentioned using an object to Kate, but looking at more of the code now you're right that would make parts slightly more complicated then they need to be.

How about letting status be an array. For example, here it would be
```
const status = _.compact([
  _.max([target_diff.length, target_explanation_diff.length]) > 1 && "target_changed",
  _.max([name_diff.length, methodology_diff.length]) > 1 && "indicator_desc_changed",
]);
```

The filtering at line 118 could be written as 
```
.filter(
  ({status}) => (status.length === 0 && indicator_status.no_diff.active) ||
    !_.chain(indicator_status)
      .pickBy('active')
      .keys()
      .union(status)
      .isEmpty()
      .value()
)
```
Could save some cycles on that by getting the active status keys just once somewhere earlier.

`get_status_flag` could also be cleaned up a lot with this, like
```
const get_status_flag = (indicator_status) => _.map(
  indicator_status,
  (status_key) => (
    <div className={`text-diff__indicator-status--${/* something here to map status key to class modifier */}`}>
      {text_maker("status")}
    </div>
  )
);
```

There may be other stuff that could be cleaned up with this too, haven't read every line in this file recently.

**Disclaimer:** I typed this all in to GitHub untested, no promises that it works as I intended it to, or that I didn't miss something in my logic

Reply:
Oh, `get_status_flag` also needs to display the empty array/"no_diff" case. That or make "no_diff" an explicit status and not just the empty array state and modify my other suggestions accordingly.

Reply:
👍